QuizList/QuestionView.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  QuestionView.swift
+//  QuizList
+//
+//  Created by Alexander v. Below on 26.08.20.
+//  Copyright © 2020 None. All rights reserved.
+//
+
+import SwiftUI
+
+extension Text {
+    public func highlight(_ highlight: Bool) -> some View {
+        if highlight {
+            return self.foregroundColor(.red)
+        } else {
+            return self
+        }
+    }
+}
+
+struct QuestionView: View {
+    var list: QuizList
+    var quizFactory: QuestionManufactory
+    @State var item = 0
+    @State var showCorrectAnswer: Int?
+
+    init(list: QuizList, item: Int? = nil) {
+        self.list = list
+        self.quizFactory = QuestionManufactory(list: list)
+        if let item = item {
+            self.item = item
+        } else {
+            self.item = quizFactory.nextQuestion()
+        }
+    }
+
+    var body: some View {
+        VStack(spacing: 10) {
+            Text("Item # \(list[item].number)").bold()
+
+            let answerSet = try! self.quizFactory.answers(question: self.item, number: 4)
+
+            ForEach(0..<answerSet.answers.count) { i in
+                Button(action: {
+                    if i == answerSet.correctAnswer {
+                        quizFactory.appendCorrectAnswer(i)
```

Comment:
Why are you appending the position of the correct answer?

Reply:
What is wrong here is that I should append `self.item`. This is done to repeat items that were wrong